### ISSUE TITLE
Sharing Does Not Share Audio

### ISSUE THREAD

**Issue by jag3773** [unfoldingWord-dev/uw-android#145]
It does not appear that sharing a resource shares the audio for that resource.

**Comment by dwsolberg** [comment]
We have not discussed how to do this. Should we just automatically include all the audio for now?

**Comment by jag3773** [comment]
Seems like it should show up in the list under the language it relates to, e.g.
```
OBS (en)
OBS Audio (en)
```

**Comment by sampeckham** [comment]
I think there a few decisions here that we or the user will want to make.
Format to share: Text, Audio, Video
Quality to share: High, Low
Quantity to share: This story/chapter/book or the entire OBS/Bible.

Personally, I feel if I'm sharing from within a chapter of OBS i would expect only that to be shared. Socially, we are used to sharing the micro, the small bit we liked, the one blog post, or even pull quote on twitter. Perhaps the context the user is in should drive the what?

**Comment by jag3773** [comment]
Likely the user will only have either High or Low quality on their device, so we could probably punt on that, right?

**Comment by sampeckham** [comment]
Yep. So sharing is essentially limited to whatever the user has already downloaded offline.

**Comment by sampeckham** [comment]
Here's the flow of screens. I've worked through the different ways this could be handled and how Android favours things. These are 'modal bottom sheets', sliding up from the bottom, the screen should have a scrim (be dimmed) when these are active. They can be exited by swiping them away or tapping outside of them.

Note: The observant will spot some other tweaks I'm working on here, not part of this issue though! : )

![screen shot 2015-11-11 at 14 29 47](https://cloud.githubusercontent.com/assets/323212/11093265/2c8fa332-8881-11e5-8511-1a2100f0a289.png)

**Comment by sampeckham** [comment]
And for iOs following the native share sheet design.

![screen shot 2015-11-11 at 15 22 15](https://cloud.githubusercontent.com/assets/323212/11094596/752837ec-8888-11e5-8347-d27df3d9d7af.png)

**Comment by PeeJWeeJ** [comment]
This looks great!

Do you have any designs for sharing/loading from the home screen?

**Comment by sampeckham** [comment]
Good question @PeeJWeeJ @jag3773  Do we have a clear spec for what it should do? Both iOS and Android are different in functionality here.

Android has Send and Receive 
iOS has only Receive

With Send we jump to a huge number of potential options at the top level. A user will have to drill down through Bible/OBS > Language > Text/Audio/Video. It could be argued therefore it's just a quick and more obvious to just open the thing you want to share and share from there - just a thought?

**Comment by PeeJWeeJ** [comment]
There is also a request for the ability to export multiple versions: https://github.com/unfoldingWord-dev/uw-android/issues/132 so we should probably have a UI for this.

Would it work to simply improve the current share layout on android by, for instance, adding some sectioning and then use the same process in the above second and third screen?

For the receive UI I was planning on using the interface from the above third screen and then going to the appropriate screen from there.

**Comment by dwsolberg** [comment]
@sampeckham This makes sense to me, at least on iOS.

**Comment by jag3773** [comment]
@sampeckham Sharing from the home screen should provide a list of resources that can be checked to be shared all at the same time.  Does that make sense?

**Comment by dwsolberg** [comment]
@jag3773 @sampeckham We don't have a UI for this, and I'm not sure it even makes sense in iOS because really only one item can reasonably be shared at a time given the file sizes. Android has a save to SD card where it makes more sense

**Comment by sampeckham** [comment]
To feed into this (it still seems undecided), but if you shared everything from the home screen, you'd have to have a UI such as below. Bear in mind this only shows OBS, so adding Bible is another layer of options and tick boxes. I'm not convinced giving all that fine grained choice is a good idea. 

I think it comes down to the question
How many use cases will there be where someone wants to share more than one translation, mobile to mobile?

Even if they want to share 2 maybe 3, it's still easy and straight forward to open the translation and tap the share option, as we have above, earlier in the thread. I think I'd be wanting to know it's happening a lot that people are shifting several translations around mobile to mobile and it's common, otherwise it feels more edge case, of one translator to another translator, rather than general user to general user. 

In addition this very quickly becomes a big file transfer too, not sure how well that will hold up over bluetooth, wifi etc. 

Anyway, here's the UI to make it happen.
Tick on the left is everything for that language.
Dash on the left is some items selected (text or audio or video, not all)
Blank nothing selected for that language.

The share button takes the user to the share options (as above) bluetooth, wifi, sd or whatever.

![uw-share-options](https://cloud.githubusercontent.com/assets/323212/11528976/ea4e1d8e-98e1-11e5-92b4-d29f6d4448b6.jpg)

**Comment by jag3773** [comment]
I like that UI and think it will do the trick.

**Comment by dwsolberg** [comment]
@sampeckham You probably know what I'm going to ask, but could I get the graphics for the empty box, ticked box, and box with the "-" -- preferably as pdfs?

**Comment by sampeckham** [comment]
Sorry, thought this was resolved on slack chat. iOS doesn't use empty boxes and checked boxes. It just has blank space or check mark on its own. Do you need the check mark, or can you call it from the standard UI one in XCode?

**Comment by dwsolberg** [comment]
@sampeckham I was thinking we'd actually want to have all the box styles in iOS on the left hand. Otherwise, things won't line up, and people may not know what the lack of a tick or "-" means. Also, I think that's how I've always seen it. Does that make sense, or am I missing something?

**Comment by dwsolberg** [comment]
@sampeckham So the share icon on the home screen will first bring up a menu about whether to send or receive, then for send, it will need provide the check box layout shown above.

**Comment by PeeJWeeJ** [comment]
@sampeckham Would it work better for users to only choose versions and have the same sharing popup for choosing audio/video when the tap share?

**Comment by dwsolberg** [comment]
This should be done on iOS.

**Comment by dwsolberg** [comment]
Whoops, there is one remaining issue: If you share twice, the validation signature isn't transmitted. Everything else should be done.

**Comment by jag3773** [comment]
Transferring the file worked on Android, but trying to open it on the receiving end crashed the app.

**Comment by jag3773** [comment]
https://github.com/unfoldingWord-dev/uw-android/issues/145#issuecomment-184434092 isn't saying much, as importing anything in build 25 causes a crash.

**Comment by PeeJWeeJ** [comment]
Should work on android in next build
